### ISSUE TITLE
Backup Categories

### ISSUE THREAD

**Issue by joe-batt** [HenriDellal/emerald#51]
When backing up the preferences, the categories and app mappings are not stored. It would be nice they could be stored with the preferences or in a separate file.

**Comment by joe-batt** [comment]
If I'm not missing something, it should be pretty easy to store in a second file. As there is one .cat file per category with just a list of the assigned applications, you could just concat them into one file with separator lines containing the category name and some leading and/or trailing character to recognize them. Unless of course android has built-in support for zip or tar or something.

If I find the time to setup a android dev environment, I'll try to build something like that.

**Comment by HenriDellal** [comment]
Yes, but there is also a properties file that should be packaged or to be replaced with notes at the start of every category. Concatenation seems too complicated for me, so zip packaging is the best choice here. Zip utilities are provided by [java.util.zip](https://docs.oracle.com/javase/7/docs/api/java/util/zip/package-summary.html) package.
Example: [Source code of a simple zip archiver](https://github.com/Coolcord/Squeez/tree/master/Squeez/src/csci567/squeez)
Part of this code might be used as a library for Emerald.

**Comment by ghost** [comment]
An 'Export Categories' would be very helpful. 
Maybe with the database what apps are inside, too.

Awesome Launcher btw. I love it! =)
